plugin.video.odeon/odeon.py

```diff
@@ -0,0 +1,589 @@
+
+    if has_subprods:
+        url = '{0}?action={1}&pid={2}&source={3}&sid={4}'.format(
+                addon_url, 'list_subprods', PID, prod['id']['source'], prod['id']['sid'])
+        is_folder = True
+    else:
+        url = '{0}?action={1}&pid={2}&source={3}&sid={4}'.format(
+                addon_url, 'play', PID, prod['id']['source'], prod['id']['sid'])
+        item.setProperty('IsPlayable', 'true') # necesario solo si se usa xbmcplugin.setResolvedUrl()
+        item.addStreamInfo('video', {'codec': 'h264'})
+        item.addStreamInfo('audio', {'codec': 'aac', 'language' : 'es'})
+        is_folder = False
+
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=item, isFolder=is_folder)
+
+
+def toggle_setview(params):
+    path = '{0}/vista/{1}/{2}'.format(params['source'], PID, params['sid'])
+    json_request(path)
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def toggle_misala(params):
+    path = '{0}/amisala/{1}/{2}'.format(params['source'], PID, params['sid'])
+    json_request(path)
+    xbmc.executebuiltin("Container.Refresh")
+
+
+def user_qualify(params):
+    # mi calificación es
+    voto = xbmcgui.Dialog().select(translation(30025), ['1', '2', '3', '4', '5'])
+    if voto == -1: return
+    path = '{0}/votar/{1}/{2}?voto={3}'.format(params['source'], PID, params['sid'], voto + 1)
+    response = requests.post(make_url(path), None, headers=get_headers(addon.getSetting('token')))
+    if response.status_code == 200:
+        xbmc.executebuiltin("Container.Refresh")
+    else:
+        xbmcgui.Dialog().ok('Error ({0})'.format(response.status_code), response.content)
+        sys.exit(0)
+
+
+def show_info(params):
+    xbmc.executebuiltin('Action(Info)')
+
+
+def sort(params):
+    # ordenar los ítems
+    order_names = [x['nom'] for x in META['order']]
+    order_ids = [x['id'] for x in META['order']]
+    index = xbmcgui.Dialog().select(translation(30026), order_names)
+    if index == -1: return
+    query = '{0}?action={1}&pid={2}&url={3}&orden={4}'.format(addon_url, 'list_prods', PID, quote(params['url']), order_ids[index])
+    xbmc.executebuiltin('Container.Update({0})'.format(query))
+
+
+def search(params):
+    # título, director, actor...
+    texto = xbmcgui.Dialog().input(translation(30027))
+    if texto == '': return
+    params['url'] = 'search/' + quote(texto)
+    list_prods(params)
+
+
+def heartbeat(source, sid, currentTime, totalTime):
+    token = addon.getSetting('token')
+    url = make_url('{0}/vista/{1}/{2}/{3}/{4}/{5}'.format(source, PID, sid, int(totalTime), int(currentTime), token))
+    r = requests.get(url, headers=get_headers())
+    # xbmc.executebuiltin('Notification(%s, %s, %d, %s)'%(addon_name, 'Beat ' + str(currentTime/60), 1000, addon_icon))
+
+
+def monitor(source, sid, seek=None):
+    # esperamos a que el player comience
+    for i in range(60):
+        if xbmc.Player().isPlayingVideo() or xbmc.abortRequested: break
```

Comment:
You should be using xbmc.Monitor.waitForAbort() ( http://mirrors.xbmc.org/docs/python-docs/15.x-isengard/xbmc.html#Monitor-waitForAbort ) possibly here and definitely below for the 3 sec wait.